pkg/plugins/builtin/httputil/httputil.go

```diff
@@ -110,3 +113,58 @@ func defaultHTTPClientFactory(cfg HTTPClientConfig) HTTPClient {
 	}
 	return c
 }
+
+func GetTransport(opts ...func(*http.Transport) error) (http.RoundTripper, error) {
+	tr := http.DefaultTransport.(*http.Transport).Clone()
+	for _, o := range opts {
+		if err := o(tr); err != nil {
+			return tr, err
+		}
+	}
+
+	_ = http2.ConfigureTransport(tr)
+	return tr, nil
+}
+
+func WithTLSInsecureSkipVerify(v bool) func(*http.Transport) error {
+	return func(t *http.Transport) error {
+		if t.TLSClientConfig == nil {
+			t.TLSClientConfig = &tls.Config{}
+		}
+
+		t.TLSClientConfig.InsecureSkipVerify = v
+		return nil
+	}
+}
+
+func WithTLSClientAuth(cert tls.Certificate) func(*http.Transport) error {
+	return func(t *http.Transport) error {
+		if t.TLSClientConfig == nil {
+			t.TLSClientConfig = &tls.Config{}
+		}
+
+		t.TLSClientConfig.Certificates = append(t.TLSClientConfig.Certificates, cert)
+		return nil
+	}
+}
+
+// WithTLSRootCA should be called only once, with multiple PEM encoded certificates as input if needed.
+func WithTLSRootCA(caCert []byte) func(*http.Transport) error {
+	return func(t *http.Transport) error {
+		if t.TLSClientConfig == nil {
+			t.TLSClientConfig = &tls.Config{}
+		}
+		caCertPool, err := x509.SystemCertPool()
```

Comment:
This code will not works on windows

Reply:
Yes, it will not work on Windows, that's why the error `if` statement just after handle the Windows case, and will have an empty pool to prevent an hard fail.